onnxruntime/core/providers/webnn/builders/impl/conv_op_builder.cc

```diff
@@ -293,22 +293,39 @@ bool ConvOpBuilder::IsOpSupportedImpl(const InitializedTensorSet& initializers,
   const auto& op_type = node.OpType();
   const auto& input_defs = node.InputDefs();
 
-  const auto& weight_name = input_defs[1]->Name();
+  std::vector<int64_t> input_shape;
+  if (!GetShape(*input_defs[0], input_shape, logger)) {
+    LOGS(logger, VERBOSE) << "Cannot get input's shape.";
+    return false;
+  }
+
+  const auto input_size = input_shape.size();
+  if (input_size != 4) {
+    LOGS(logger, VERBOSE) << op_type << " [" << name << "]'s input dimension: " << input_size
+                          << ". Only conv 2d is supported.";
+    return false;
+  }
+
+  std::vector<int64_t> weight_shape;
+  if (!GetShape(*input_defs[1], weight_shape, logger)) {
+    LOGS(logger, VERBOSE) << "Cannot get weight's shape.";
+    return false;
+  }
+
+  const auto weight_size = weight_shape.size();
+  if (weight_size != 4) {
+    LOGS(logger, VERBOSE) << op_type << " [" << name << "]'s weight dimension: " << weight_size
+                          << ". Only conv 2d is supported.";
+    return false;
+  }
+
   // WebNN CPU backend (XNNPACK) requires the filter operand to be a constant.
```

Comment:
I hope XNNPack can fix this in the future...

Reply:
I hope so. :)